src/main/java/com/teamabnormals/caverns_and_chasms/common/item/ImpactPotionItem.java

```diff
@@ -0,0 +1,97 @@
+import net.minecraft.util.StringUtil;
+import net.minecraft.world.effect.MobEffect;
+import net.minecraft.world.effect.MobEffectInstance;
+import net.minecraft.world.entity.EquipmentSlot;
+import net.minecraft.world.entity.ai.attributes.Attribute;
+import net.minecraft.world.entity.ai.attributes.AttributeModifier;
+import net.minecraft.world.item.*;
+import net.minecraft.world.item.alchemy.PotionUtils;
+import net.minecraft.world.level.Level;
+import net.minecraftforge.api.distmarker.Dist;
+import net.minecraftforge.api.distmarker.OnlyIn;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.List;
+import java.util.Map;
+
+public class ImpactPotionItem extends TetherPotionItem {
+
+	public ImpactPotionItem(Properties properties) {
+		super(properties);
+	}
+
+	@OnlyIn(Dist.CLIENT)
+	@Override
+	public void appendHoverText(ItemStack stack, @Nullable Level level, List<Component> tooltip, TooltipFlag flag) {
+		List<MobEffectInstance> list = PotionUtils.getMobEffects(stack);
+		List<Pair<Attribute, AttributeModifier>> list1 = Lists.newArrayList();
+
+		List<Component> instanttooltip = Lists.newArrayList();
+		List<Component> continuoustooltip = Lists.newArrayList();
+
+		if (list.isEmpty()) {
+			continuoustooltip.add((Component.translatable("effect.none")).withStyle(ChatFormatting.GRAY));
+		} else {
+			for (MobEffectInstance mobeffectinstance : list) {
+				MutableComponent mutablecomponent = Component.translatable(mobeffectinstance.getDescriptionId());
+				MobEffect effect = mobeffectinstance.getEffect();
+				Map<Attribute, AttributeModifier> map = effect.getAttributeModifiers();
+				if (!map.isEmpty()) {
+					for (Map.Entry<Attribute, AttributeModifier> entry : map.entrySet()) {
+						AttributeModifier attributemodifier = entry.getValue();
+						AttributeModifier attributemodifier1 = new AttributeModifier(attributemodifier.getName(), effect.getAttributeModifierValue(mobeffectinstance.getAmplifier(), attributemodifier), attributemodifier.getOperation());
+						list1.add(new Pair<>(entry.getKey(), attributemodifier1));
+					}
+				}
+
+				if (mobeffectinstance.getAmplifier() > 0) {
+					mutablecomponent = Component.translatable("potion.withAmplifier", mutablecomponent, Component.translatable("potion.potency." + mobeffectinstance.getAmplifier()));
+				}
+
+				if (effect.isInstantenous()) {
+					instanttooltip.add(mutablecomponent.withStyle(effect.getCategory().getTooltipFormatting()));
+				} else {
+					mutablecomponent = Component.translatable("potion.withDuration", mutablecomponent, StringUtil.formatTickDuration(mobeffectinstance.getDuration()));
+					continuoustooltip.add(mutablecomponent.withStyle(effect.getCategory().getTooltipFormatting()));
+				}
+			}
+		}
+
+		tooltip.addAll(instanttooltip);
+		if (!continuoustooltip.isEmpty()) {
+			tooltip.add(Component.empty());
+			tooltip.add((Component.translatable("item.modifiers.splash").withStyle(ChatFormatting.GRAY)));
```

Comment:
Change this to potion.whenSmashed actually to match potion.whenDrank I think. In my opinion I'd recommend making methods that references any of the non-duplicated code between Tether Potion's method for appendHoverText so not as much needs to be duplicated